context-free.md

```diff
@@ -1,21 +1,39 @@
 # Context Free
 
 ##  Who is this programming language for?
-
+This programming language is used to create images, or more specifically, create complex images using only variations of
+	circles, triangles, and squares. Combining with the ability to render images with size up to 100 mega pixels, I believe 
+	the program is directed towards people who liked to design but just not o nthe professional level yet. 
 
 ## What is easy to do in this language? Why is it easy?
-
+Like said above, this program has the ability to manipulate a large amount of shapes, allowing people to create images 
+	only defined by their coding ability and tehir imagination. Also, as a DSL, Context Free excels in handling programs 
+	written to create images only and also by using precise "expressions" that is specific to this area. 
 
 ## What is hard to do in this language? Why is it hard?
-
+Due to the properties of Context Free and DSLs, their scope of use is relatively narrow, many are designed specificlly to
+	complete directions specific for one category. This meant that it is hard for Context Free to complete things outsisw of 
+	its scope. One challenge when using Context Free is that the program shapes is limited to circles, squares, and triangles,
+	which meant that it is hard for one to draw something like a detailed car, or other things with irregular shapes. In short,
+	the program excels in completing the tasks it is designed for, but not for completing other actions.
 
 ## How did you learn how to program in this language?
-
+I learned how to program in this language by using the github wiki on Context Free to learn about the language, uses of each
+	codem and also the appropriate time to use it. I also took ideas and also concepts from the example program to strengthen my
+	understanding.
 
 ## Describe what happens when a ContextFree program runs
 _Specifically, how does the computer interpret the rules in a program, in
 order to turn those rules into a picture?_
+When a Context Free program runs, the program is similar to a tree. The program takes the basic shapes of cirle, square, and 
```

Comment:
I think your explanation is mostly correct. I like your analogy that compares the program with the tree. In fact, it helps me to understand more thoroughly what happen when the program runs. And I agree your last sentence also; indeed context-free is able to create extremely complex shapes with just few lines of code.